### DISCUSSION TITLE
Use HVM in a non-interactive bash script on Ubuntu

### DISCUSSION THREAD

**Discussion by bowman2001** [jmooring/hvm#69] [Show and tell]
The following special circumstances gave me a little headache today. I’m not sure if this special case is worth mentioning in the README. But hopefully others will find this before it takes them a little while to figure it out.

## Context
- I ran hugo in a bash script on a recent Ubuntu machine
- I did something like `source .bashrc` to include the HVM functions I put there
- But the script did not use HVM at all

## Problem
The reason was the following short-circuit in Ubuntu’s default `.bashrc`:

```bash
# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac
```
## Solution

The HVM functions generatd by `hvm gen alias bash` need to be placed before this statement in `.bashrc`.

**Comment by jmooring** [comment]
> The reason was the following short-circuit in Ubuntu’s default .bashrc:

Yeah, you look at that bit and say, "That makes sense."  And then you ssh into the system and can't figure out why your aliases, paths, etc. aren't working. And even though you are interacting with the system interactively, you are not "running interactively."

I think everyone gets kicked by this... once.
